packages/kbn-es/src/utils/docker.ts

```diff
@@ -0,0 +1,394 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import chalk from 'chalk';
+import execa from 'execa';
+import fs from 'fs';
+import Fsp from 'fs/promises';
+import { resolve } from 'path';
+
+import { ToolingLog } from '@kbn/tooling-log';
+import { kibanaPackageJson as pkg } from '@kbn/repo-info';
+
+import { createCliError } from '../errors';
+import { EsClusterExecOptions } from '../cluster_exec_options';
+
+interface BaseOptions {
+  tag?: string;
+  image?: string;
+}
+
+export interface DockerOptions extends EsClusterExecOptions, BaseOptions {
+  dockerCmd?: string;
+}
+
+export interface ServerlessOptions extends EsClusterExecOptions, BaseOptions {
+  clean?: boolean;
+  basePath: string;
+}
+
+interface ServerlessEsNodeArgs {
+  esArgs?: Array<[string, string]>;
+  image: string;
+  name: string;
+  params: string[];
+}
+
+const DOCKER_REGISTRY = 'docker.elastic.co';
+
+const DOCKER_BASE_CMD = [
+  'run',
+
+  '--rm',
+
+  '-t',
+
+  '--net',
+  'elastic',
+
+  '--name',
+  'es01',
+
+  '-p',
+  '127.0.0.1:9200:9200',
+
+  '-p',
+  '127.0.0.1:9300:9300',
+];
+
+const DEFAULT_DOCKER_ESARGS: Array<[string, string]> = [
+  ['ES_JAVA_OPTS', '-Xms1g -Xmx1g'],
```

Comment:
Should these be `Xms1536m` to match the other defaults?  I'm guessing it was upped to support larger snapshot testing at some point.

Reply:
Added fb1e8b9dfa222f900a64999a85bfc5f0e5e0fba7.

For serverless the docs recommended `Xms1g` per node, so I kept that.